### ISSUE TITLE
NoClassDefFoundError on older versions of Android

### ISSUE THREAD

**Issue by sesterki** [onnela-lab/beiwe-android#43]
We got this error on a phone running android 4.4.2 using Version 31 of the code base
```
java.lang.ClassNotFoundException: Didn't find class "android.graphics.drawable.Icon" on path: DexPathList[[zip file "/data/app/org.beiwe.app-1.apk"],nativeLibraryDirectories=[/data/app-lib/org.beiwe.app-1, /vendor/lib, /system/lib]]
    at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:497)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:457)
    at java.lang.Class.getDeclaredMethods(Class.java)
    at java.lang.Class.getDeclaredMethods(Class.java:656)
    at android.view.ViewDebug.getExportedPropertyMethods(ViewDebug.java:960)
    at android.view.ViewDebug.exportMethods(ViewDebug.java:1047)
    at android.view.ViewDebug.dumpViewProperties(ViewDebug.java:997)
    at android.view.ViewDebug.dumpViewProperties(ViewDebug.java:983)
    at android.view.ViewDebug.dumpView(ViewDebug.java:900)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:870)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dump(ViewDebug.java:793)
    at android.view.ViewDebug.dispatchCommand(ViewDebug.java:416)
    at android.view.ViewRootImpl$W.executeCommand(ViewRootImpl.java:6258)
    at android.view.IWindow$Stub.onTransact(IWindow.java:65)
    at android.os.Binder.execTransact(Binder.java:404)
    at dalvik.system.NativeStart.run(NativeStart.java)
java.lang.NoClassDefFoundError: android/graphics/drawable/Icon
    at java.lang.Class.getDeclaredMethods(Class.java)
    at java.lang.Class.getDeclaredMethods(Class.java:656)
    at android.view.ViewDebug.getExportedPropertyMethods(ViewDebug.java:960)
    at android.view.ViewDebug.exportMethods(ViewDebug.java:1047)
    at android.view.ViewDebug.dumpViewProperties(ViewDebug.java:997)
    at android.view.ViewDebug.dumpViewProperties(ViewDebug.java:983)
    at android.view.ViewDebug.dumpView(ViewDebug.java:900)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:870)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dumpViewHierarchy(ViewDebug.java:867)
    at android.view.ViewDebug.dump(ViewDebug.java:793)
    at android.view.ViewDebug.dispatchCommand(ViewDebug.java:416)
    at android.view.ViewRootImpl$W.executeCommand(ViewRootImpl.java:6258)
    at android.view.IWindow$Stub.onTransact(IWindow.java:65)
    at android.os.Binder.execTransact(Binder.java:404)
    at dalvik.system.NativeStart.run(NativeStart.java)
```

**Comment by sesterki** [comment]
This issue occurs for older versions of Android (e.g. 4.4.2) trying to display notifications after our API 26 update. I have added some code to work around this in 09c59a40c9499bb0a2c0acae3888565652f389cc

**Comment by zagorsky** [comment]
@sesterki fixed this in 09c59a4, app version 2.4.1 and above.
